File: typescript/src/util/check.ts
export interface Validator<T> {
    (value: T): string
}

export class Check {

    static required<T> (property: string, value: T, ...validators: Validator<T>[]): T {
        if (value == null || value == undefined) {
            throw new Error(`${property} is required`);
        }
        if (typeof value == 'string' && value.length == 0) {
            throw new Error(`${property} must not be empty`);
        }
        for (const validator of validators) {
            const error = validator(value);
            if (error) {
                throw new Error(`${property}: '${value}' ${error}`);
            }
        }
        return value;
    }

    static min(min: number): Validator<number> {
        return value => (value < min) ? `must be at least ${min}` : null;
    }

    static format(regex: RegExp, error: string = 'invalid format'): Validator<string> {
        return value => !value.match(regex) ? error : null;
    }
}

File: python/util/check.py
import re


def required(property_name, value, *validators):
    if value is None:
        raise Exception(f'\'{property_name}\' is required')
    if isinstance(value, str) and len(value) == 0:
        raise Exception(f'\'{property_name}\' must not be empty')
    for validator in validators:
        error = validator(value)
        if error:
            raise Exception(f'\'${property_name}\': \'${value}\' ${error}')
    return value


def check(value, condition, error_message):
    if not condition(value):
        raise Exception(error_message)


def is_number():
    return lambda value: check(value, lambda v: isinstance(value, int) or isinstance(value, float), 'must be a number')


def minimum(min_value):
    return lambda value: check(value, lambda v: v >= min_value, f'must be at least {min_value}')


def has_format(regex, error_message):
    return lambda value: check(value, lambda v: re.match(regex, v), error_message)
